packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.resolver.ts

```diff
@@ -127,4 +135,52 @@ export class FieldMetadataResolver {
       fieldMetadataGraphqlApiExceptionHandler(error);
     }
   }
+
+  @ResolveField(() => RelationDTO, { nullable: true })
+  async relation(
+    @AuthWorkspace() workspace: Workspace,
+    @Parent() fieldMetadata: FieldMetadataEntity<FieldMetadataType.RELATION>,
+    @Context() context: { loaders: IDataloaders },
+  ): Promise<RelationDTO | null | undefined> {
+    if (!isRelationFieldMetadataType(fieldMetadata.type)) {
+      return null;
+    }
+
+    const isNewRelationEnabled = await this.featureFlagService.isFeatureEnabled(
+      FeatureFlagKey.IsNewRelationEnabled,
+      workspace.id,
+    );
+
+    if (!isNewRelationEnabled) {
+      throw new BadRequestException(
+        'New relation feature is not enabled for this workspace',
+      );
+    }
+
+    try {
+      const {
+        sourceObjectMetadata,
+        targetObjectMetadata,
+        sourceFieldMetadata,
+        targetFieldMetadata,
+      } = await context.loaders.relationLoader.load({
+        fieldMetadata,
+        workspaceId: workspace.id,
+      });
+
+      if (!fieldMetadata.settings) {
+        throw new BadRequestException('Relation settings are required');
```

Comment:
a resolver should throw a GraphqlError directly (not a BadRequestException which is HTTP / Rest standard)

Reply:
or as you are wrapping in a tryCatch, this could be a custom exception too (Metadata related)